### PR TITLE
pessimistic locks use MAX_TTL

### LINKED ISSUE #328: Transaction::lock_keys should wait instead of return a `Failed to resolve lock` Error.
Transaction::lock_keys should wait instead of return a `Failed to resolve lock` Error. As other language's standard library `lock` call will block current thread or coroutine. So Transaction::lock_keys should block until got the lock. And we can provide a `try_lock_keys` function that return immediately.

### PR DESCRIPTION
**Pull request by ekexium**
close #328 

Pessimistic locks now uses the default ttl(3s). The heartbeat mechanism by default works with the interval of 10s. It cannot keep pessimistic locks alive. Change it to MAX_TTL(20s).

### CODE REVIEW

**Comment by blacktear23** [comment]
When will this PR can be merged?

**Comment by ekexium** [comment]
@blacktear23 Awaiting at least 1 approval from reviewers with write access.
ping @andylokandy

**Review by andylokandy — APPROVED** [review]

**Pull request merged**
